snapcraft/tests/test_lifecycle.py

```diff
@@ -169,6 +172,68 @@ def test_dirty_strip_restrips(self):
             'Stripping part1 \n',
             self.fake_logger.output)
 
+    def test_dirty_stage_restages(self):
```

Comment:
I updated the strip and stage tests here to have multiple parts, with two tests each-- where one part is dirty (with the other built) and where both parts are dirty.